### ISSUE TITLE
`Toplevel` is confused about dimension - Should use `Computed` and `Dim.Fill` 

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#3135]
In investigating 

- #3127 (in PR #3130)

I found that `Toplevel` doesn't actually work the way we think it does. For example:

```cs
/// <summary>
/// Convenience factory method that creates a new Toplevel with the current terminal dimensions.
/// </summary>
/// <returns>The created Toplevel.</returns>
public static Toplevel Create () => new Toplevel (new Rect (0, 0, Driver.Cols, Driver.Rows));

/// <summary>
/// Initializes a new instance of the <see cref="Toplevel"/> class with the specified <see cref="LayoutStyle.Absolute"/> layout.
/// </summary>
/// <param name="frame">A Superview-relative rectangle specifying the location and size for the new Toplevel</param>
public Toplevel (Rect frame) : base (frame) => SetInitialProperties ();

/// <summary>
/// Initializes a new instance of the <see cref="Toplevel"/> class with <see cref="LayoutStyle.Computed"/> layout, 
/// defaulting to full screen.
/// </summary>
public Toplevel () : base ()
{
	SetInitialProperties ();
	Width = Dim.Fill ();
	Height = Dim.Fill ();
}

static bool SetCurrentOverlappedAsTop()
{
    if (OverlappedTop == null && Current != Top && Current?.SuperView == null && Current?.Modal == false)
    {
        if (Current.Frame != new Rect(0, 0, Driver.Cols, Driver.Rows))
        {
            // BUGBUG: Use Dim.Fill
            Current.Frame = new Rect(0, 0, Driver.Cols, Driver.Rows);
        }
        Top = Current;
        return true;
    }
    return false;
}
```

Note `Create` and `SetCurrentOverlappedAsTop` use absolute positioning and the constructor uses `Computed` (`Dim.Fill ()`).

All of this will likely be changed when we get to:

- #2502
- #2491
- #2537
- #272
- #2407 

However, I'd like to fix this now so that #3130 can be unblocked. I'll submit a PR for this asap...

- [ ] Refactor unit tests to be able to more clearly focus on Toplevel
- [ ] Change `Create` to use `Dim.Fill`
- [ ] Fix `SetCurrentOverlappedAsTop` related issues.
- [ ] Fix other code that depends on old bad behavior
- [ ] Fix related unit tests that assumed old bad behavior

**Comment by tig** [comment]
In tracking down unit test failures found after changing `Toplevel` to use `Dim.Fill` I noticed this:

```cs
public static void Init (ConsoleDriver driver = null, string driverName = null) => InternalInit (Toplevel.Create, driver, driverName);
```

When I simplified `Init` for v2 way back, the `public` ability to pass a top-level factory to `Init` was removed. 

However, I failed to remove the whole `toplevelFactory` from `InternalInit` as well.

In this PR, I'd like to:

- Change `InternalInit` to not take `toplevelFactory`
- Remove `Toplevel.Create`
- Have `Application.InternalInit` call `SetRelativeLayout` before it returns

This will

a) Further simplify `Application.Init` logic without really impacting the public API (nobody but unit tests use `Toplevel.Create` because there's  no point.
b) Make it so existing unit tests that depended on `Application.Top` having valid `Dim`s pass without modification.

Any objections?
